### PR TITLE
fix read more link issue

### LINKED ISSUE #6401: "Read more" links sometimes appear when not needed
Sometimes "Read more" and "Read less" links are present below a completely visible block of content.  In cases such as these, we should not display the links.

### Evidence / Screenshot (if possible)
useless_readmore

### Relevant url?

"Read more" links can be found on our book pages.  Here is a link to the page pictured above: https://openlibrary.org/works/OL52430W/Peer_Gynt

### Steps to Reproduce

1. Navigate to a page where this issue is occurring (the link above is a good place to start).
2. Click the "Read more" link.

* Actual: The "Read more" link is replaced with a "Read less" link, but no new content is displayed.
* Expected: The "Read more" link is replaced with a "Read less" link, and new content is present on the screen.

### Details

- **Logged in (Y/N)?** N
- **Browser type/version?** FF 97.0.2
- **Operating system?** Ubuntu
- **Environment (prod/dev/local)?** prod

### Proposal & Constraints

There is already Javascript that hides the "Read more" link if the content is under a specific, hard-coded height.  Maybe changing this to compare the content's scroll height with its actual height would correct the issue?  We are doing something similar with our subject clamping code.

### Related files

Javascript for the "Read more" links and subject clamping are located in the same file:
https://github.com/internetarchive/openlibrary/blob/master/openlibrary/plugins/openlibrary/js/readmore.js

Read more link and subject clamper code is imported and initialized here (no changes are necessary here --- this is just for reference):
https://github.com/internetarchive/openlibrary/blob/46bcf7c30178f53ef427a5840c4b24717ed79fb0/openlibrary/plugins/openlibrary/js/index.js#L198-L211

### PR DESCRIPTION
**Pull request by tangym27**
Closes #6401 

- Fix such that the "read more/less" button does not appear
- Button will also appear/disappear when the screen gets resized

### Technical

- Resizing based off of code from `SearchBar.js`
- Hardcoded `58` is based off of the height of `.restricted-height` (made a comment in the code itself)

### Testing

(Locally): [http://localhost:8080/works/OL13101191W/Alice%27s_Adventures_in_Wonderland](http://localhost:8080/works/OL13101191W/Alice%27s_Adventures_in_Wonderland
)
### Screenshot

With resizing in action (sorry for low quality):
read_more

### Stakeholders

@jimchamp

### CODE REVIEW

**Comment by mekarpeles** [comment]
@tangym27, this looks like a great direction to me, stellar job on this PR and thank you for including the screencast!

**Comment by mekarpeles** [comment]
Tests are failing on coverage (which is OK) we just need to change the coverage % here: 
https://github.com/internetarchive/openlibrary/blob/master/package.json#L99-L104

File: openlibrary/plugins/openlibrary/js/readmore.js [hunk 1]
```diff
@@ -11,13 +30,7 @@ export function initReadMoreButton() {
...
         $(`.${up.attr('class')}.read-less`).addClass('hidden');
     });
-    $('.restricted-view').each(function() {
-        if ($(this).outerHeight()<50) {
-            $(`.${$(this).parent().attr('class')}.read-more`).addClass('hidden');
-        } else {
-            $(this).addClass('restricted-height');
-        }
-    });
+    $(window).on('resize', debounce(resetReadMoreButtons, 50));
```
Comment (cdrini) [inline openlibrary/plugins/openlibrary/js/readmore.js:33]: One extra deleted here :)

```suggestion
    resetReadMoreButtons();
    $(window).on('resize', debounce(resetReadMoreButtons, 50));
```

Comment (mekarpeles) [inline openlibrary/plugins/openlibrary/js/readmore.js:33]:
```suggestion
    resetReadMoreButtons();
    $(window).on('resize', debounce(resetReadMoreButtons, 50));
```

**Review by cdrini — APPROVED** [review]
Awesome, lgtm! Tested on testing:

- Book with short descr: https://testing.openlibrary.org/works/OL82563W/Harry_Potter_and_the_Philosopher's_Stone?edition=ia%3Aharrypotterphilo0000rowl_j1k7
- Book with long descr: https://testing.openlibrary.org/works/OL81589W/Lisey's_Story?edition=ia:liseysstorynovel00king_0

**Pull request merged**